library.rb

```diff
@@ -1,3 +1,110 @@
+require './author'
+require './book'
+require './reader'
+require './order'
+
 class Library
+  REQUIRED_COLLECTIONS = %w(Author Book Reader Order)
+  attr_reader :authors, :books, :readers, :orders
+  def initialize
+    @authors = []
+    @books = []
+    @readers = []
+    @orders = []
+    puts 'Use :load method for load library from file.'
+  end
+
+  def load
+    REQUIRED_COLLECTIONS.each do |name|
+      load_from_file(name)
+    end
+  end
+
+  def save
+    REQUIRED_COLLECTIONS.each do |name|
+      save_to_file(name)
+    end
+  end
+
+# for interface
+  def new_author(params)
+    new_class_instance("Author", @authors, params)
+  end
+
+  def new_book(params)
+    unless @authors.map(&:name).index(params.last)
+      puts "Library does not have the author #{params.last}"
+      return
+    end
+    new_class_instance("Book", @books, params)
+  end
+
+  def new_reader(params)
+    new_class_instance("Reader", @readers, params)
+  end
+
+  def new_order(params)
+    unless @books.map(&:title).index(params.first)
+      puts "Library does not have the book #{params.first}"
+      return
+    end
+    unless @readers.map(&:name).index(params[1])
+      puts "#{params[1]} wasn't registered at this library"
+      return
+    end
+    new_class_instance("Order", params)
+  end  
+
+  def best_reader
+    group_by_reader = @orders.group_by { |e| e.reader }
+    sort_readers = group_by_reader.keys.sort { |x,y| group_by_reader[y].count <=> group_by_reader[x].count }
```

Comment:
1. Why do you sort keys? You can just do something like group_by_reader.sort_by { |reader, orders| orders.size }.last or first
2. Why do you need sort_readers.select { |e| group_by_reader[e].count == group_by_reader[sort_readers.first].count } ?

You increase complexity using additional cycles and extra variables.